### ISSUE TITLE
"nocomma" shouldn't apply to array/object literals and parameter lists

### ISSUE THREAD

**Issue by Krinkle** [jshint/jshint#2044]
Introduced in #1876, this looks broken. It's emitting warnings for commas in e.g. object literals. I found this by enabling the new `enforceall` option, which is practically unusable for any serious project as long as it includes `nocomma` in its current state.

```json
{
  "nocomma": true
}
```
```js
function foo(d, e) {
	return {
		a: 1,
		b: 2,
		c: ['x', 'y'],
		d: e,
		e: e
	};
}
foo();
```

```
      1 |function foo(d, e) {
                       ^ Unexpected use of a comma operator.
      3 |        a: 1,
                     ^ Unexpected use of a comma operator.
      4 |        b: 2,
                     ^ Unexpected use of a comma operator.
      5 |        c: ['x', 'y'],
                        ^ Unexpected use of a comma operator.
      5 |        c: ['x', 'y'],
                              ^ Unexpected use of a comma operator.
      6 |        d: e,
                     ^ Unexpected use of a comma operator.
```

**Comment by impinball** [comment]
Confirmed here, too.

**Comment by impinball** [comment]
@Krinkle Could I recommend you change the title to also include parameter lists?

**Comment by kuus** [comment]
+1

**Comment by jhinrichsen** [comment]
+1

**Comment by jugglinmike** [comment]
In general, "+1" isn't particularly helpful, but you'll all be glad to know that the fix is in master (33612f8cedce779d0d22571b72da4216d18bee9c) and will be included in the very next release of JSHint

**Comment by jugglinmike** [comment]
This was included in [the 2.6.1 release of JSHint](https://github.com/jshint/jshint/releases/tag/2.6.1) back in February of this year. Thanks for the report!
